lms/djangoapps/courseware/views.py

```diff
@@ -1581,8 +1597,7 @@ def financial_assistance_form(request):
             Q(_expiration_datetime__isnull=True) | Q(_expiration_datetime__gt=datetime.now(UTC())),
             course_id=enrollment.course_id,
             mode_slug=CourseMode.VERIFIED
-        ).exists()
-        and enrollment.mode != CourseMode.VERIFIED
+        ).exists() and enrollment.mode != CourseMode.VERIFIED
```

Comment:
I just checked with the good folks at #python IRC, they said that if the first condition is False the second one won't be executed, therefor I'm moving this to the first position. Nice catch!